pkg/operator/sync_common.go

```diff
@@ -81,26 +85,68 @@ func manageControllerResources(c serviceCAOperator, resourcePath string, modifie
 // TODO manage rotation in addition to initial creation
 func manageSignerCA(client coreclientv1.SecretsGetter, eventRecorder events.Recorder) (bool, error) {
 	secret := resourceread.ReadSecretV1OrDie(v4_00_assets.MustAsset("v4.0.0/service-serving-cert-signer-controller/signing-secret.yaml"))
-	_, err := client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
+	var (
+		caCertPem, caKeyPem, signedByOldPem, fullBundlePem []byte
+	)
+
+	existing, err := client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
 	if !apierrors.IsNotFound(err) {
-		return false, err
+		if err != nil {
+			return false, err
+		}
+		// If we're not yet half expired, return.
+		certs, err := cert.ParseCertsPEM(existing.Data["tls.crt"])
+		if err != nil {
+			return false, err
+		}
+		if certs[0] != nil {
+			if !util.CertHalfwayExpired(certs[0]) {
```

Comment:
This seems a little long, isn't it? What about one month? something similar to how certmonger checks before doing a certificate renewal

Reply:
We need a longer period of time so that we can be sure that all payloads using the CA were restarted and thus picked the new bundle